### ISSUE TITLE
Authentication does not seem to be working quite right

### ISSUE THREAD

**Issue by mattcarrier** [dnephin/dobi#82]
First Off, awesome tool...the amount of scripting that I can replace with this tool is amazing!

When I try to build an image that requires one of our private images as a base image the authentication is not working:

```bash
$ docker login -u <my-user> -p <my-pass> <my-url>
Login Succeeded
$ dobi
Step 1/10 : FROM <company>/<project>/alpine-docker:17.03.0-ce
[ERROR] failed to execute task "<task>:build": Get <image-url>: unknown: Authentication is required
$ docker pull <company>/<project>/alpine-docker:17.03.0-ce
17.03.0-ce: Pulling from <project>/alpine-docker
ec37562cf8fa: Already exists 
371cc900ff27: Pull complete 
51ad1472b975: Pull complete 
76c902807740: Pull complete 
Digest: sha256:c5ca1b30d6731ac39df368936b03a83824be99d98fb202a12beb55ad7b4ac541
Status: Downloaded newer image for <company>/<project>/alpine-docker:17.03.0-ce
````

dobi.yaml
```yaml
meta:
  project: <project name>
  default: build

image=<image name>:
  image: <company>/<project>/<image name>
  context: <directory>
  tags:
    - 'latest'
    - '{git.short-sha}'
  description: "Build the <image name> image"

alias=build:
  tasks: ['<image name>:build']
  description: "Build images"

alias=tag:
  tasks: ['<image name>:tag']
  description: "Build and tag images"

alias=push:
  tasks: ['<image name>:push']
  description: "Build, tag, and push images"

alias=clean:
  tasks: ['<image name>:rm']
  description: "Remove all the images"
```

My question is why is dobi not using the same auth mechanism as docker? And if it is then it doesn't seem to be exactly the same.

**Comment by dnephin** [comment]
Hmm, I thought I had fixed this. Which version of `dobi` are you using?

`dobi` should be using the same config file for auth. The code used by the docker client wasn't easy to import because of the dependency tree, but maybe that has improved since I last tried.

**Comment by mattcarrier** [comment]
Versions 0.8 and 0.9 produced the same results.  If there is a newer release you would like me to try just point me at it and I'll give it a go.

**Comment by mattcarrier** [comment]
Could this commit contain the fix?
e49e038d1112fde85b81842cec599a4d7d538b40

If so, would it be possible to cut 0.9.1 release?

**Comment by dnephin** [comment]
Would you mind providing me with a couple pieces of data so I can create a test case? 

* an example of `<company>/<project>/<image name>` that has the same special characters (slashes, dots, colons, etc). The names can be different, but the special characters may affect the parsing
* The related entry from `~/.docker/config.json`, without the values for auth. I'm just wondering if the key has any scheme, or port, etc

**Comment by mattcarrier** [comment]
Absolutely:

x = alpha
xxxxxxxxxx-xxxxxx.jfrog.io/xxxx/alpine-docker:17.03.0-ce
xxxxxxxxxx-xxxxxx.jfrog.io/xxxx/ci-image-acceptance-core

```json
{
	"auths": {
		"http://xxxxxxxxxx-xxxxxx.jfrog.io": {
			"auth": "<my-auth>"
		},
		"https://xxxxxxxxxx-xxxxxx.jfrog.io/v2": {
			"auth": "<my-auth>"
		},
		"xxxxxxxxxx-xxxxxx.jfrog.io": {
			"auth": "<my-auth>"
		}
	}
}
```

Thanks again!

**Comment by dnephin** [comment]
I wrote a couple test cases for parsing those image names, and they both appear to return `xxxxxxxxxx-xxxxxx.jfrog.io`. 

Looking at the error text again, I realize the problem is that the private image is in the `FROM` line of a build. That's different from the previous issues. I think the auth needs to be passed to build. I'll look into how that is done.

**Comment by mattcarrier** [comment]
@dnephin Thanks for the quick turn-around

I created a Release for v0.9.1 here with this fix included while @dnephin works on 0.10:
https://github.com/Intersections/dobi/releases/tag/v0.9.1

Installation:
```bash
curl -L -o /usr/local/bin/dobi "https://github.com/Intersections/dobi/releases/download/v0.9.1/dobi-$(uname -s)"; chmod +x /usr/local/bin/dobi
```
